### PR TITLE
add libgeotiff/1.5.1

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **libgeotiff/1.5.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

https://github.com/conan-io/conan-center-index/issues/1023

### CODE REVIEW

**Comment by SpaceIm** [comment]
Few comments about patch before you ask:
- libjpeg and zlib aren't direct dependencies of libgeotiff, but transitive dependencies coming from libtiff. Original libgeotift's CMakeLists is very fragile regarding dependencies definitions and link property.
- CXX linker is used if shared because proj is a C++ library with C API. Otherwise link step (during build) fails at least on MacOS if proj is static.

File: recipes/libgeotiff/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "doc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GeoTIFF"
+        self.cpp_info.names["cmake_find_package_multi"] = "GeoTIFF"
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("m")
```
Comment (uilianries) [inline recipes/libgeotiff/all/conanfile.py:74]:
```suggestion
            self.cpp_info.system_libs = ["m", "pthread", "dl"]
```

libgeotiff requires all these 3 libs

File: recipes/libgeotiff/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        if not self.options.shared or self.options["proj"].shared: # because PROJ is a C++ library with C API
+            del self.settings.compiler.libcxx
```
Comment (uilianries) [inline recipes/libgeotiff/all/conanfile.py:36]:
```suggestion
            del self.settings.compiler.libcxx
            del self.settings.compiler.cppstd
```

File: recipes/libgeotiff/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=os.path.join(self._source_subfolder, "libgeotiff"))
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "doc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GeoTIFF"
```
Comment (uilianries) [inline recipes/libgeotiff/all/conanfile.py:70]: Interesting, the generated cmake config file is all lower-case, but the target is GeoTIFF.

File: recipes/libgeotiff/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:74]: You're right, I also see pthread and dl, but I don't see anything in source code related to dl and pthread.
So, I've compiled libgeotiff and all dependencies of the tree from source code as shared libs. I appears that:
- libtiff also depends on pthread and dl and it's not listed in its recipe.
- xz_utils depends on pthread and it's also not listed.

File: recipes/libgeotiff/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:36]: libgeotiff is a C library, there is no reason to keep cppstd I think.

Comment (uilianries) [reply recipes/libgeotiff/all/conanfile.py:36]: keep? my comment is about deleting cppstd

Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:36]: It's already deleted above without conditions ;)

Comment (uilianries) [reply recipes/libgeotiff/all/conanfile.py:36]: :man_facepalming:

File: recipes/libgeotiff/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:74]: another test:

```bash
conan create -k . libgeotiff/1.5.1@ -o libgeotiff:shared=True -o proj:shared=False -o proj:threadsafe=False -o sqlite3:shared=False -o libtiff:shared=False -o libtiff:jbig=False -o libtiff:jpeg=False -o libtiff:webp=False -o libtiff:lzma=False -o libtiff:zlib=False -o libtiff:zstd=False -b
```

```bash
ldd libgeotiff.so
	linux-vdso.so.1 (0x00007fff2499f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fc2a8a82000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc2a86af000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc2a8311000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc2a80f9000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc2a7d08000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc2a9333000)
```

Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:74]:
```
conan create -k . libgeotiff/1.5.1@ -o libgeotiff:shared=True -o proj:shared=False -o proj:threadsafe=False -o sqlite3:shared=False -o sqlite3:omit_load_extension=True -o libtiff:shared=False -o libtiff:jbig=False -o libtiff:jpeg=False -o libtiff:webp=False -o libtiff:lzma=False -o libtiff:zlib=False -o libtiff:zstd=False -b
```

```
ldd libgeotiff.so
	linux-vdso.so.1 (0x00007ffe22796000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f6cedb32000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6ced794000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6ced57c000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6ced18b000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6cee5af000)
```

Comment (uilianries) [reply recipes/libgeotiff/all/conanfile.py:74]: nice investigation!

File: recipes/libgeotiff/all/conanfile.py (hunk 2 above)
Comment (danimtb) [inline recipes/libgeotiff/all/conanfile.py:36]: Why is this deleted conditionally to the shared option?

Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:36]: because libgeotiff is a C library, while proj is a C++ library. But anyway I need to remove `self.options["proj"].shared` because it's unknown in `configure`

Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py:36]: I will be able to unconditionally remove libcxx in libgeotiff recipe with these PR https://github.com/conan-io/conan-center-index/pull/1529 and https://github.com/conan-io/conan-center-index/pull/1530.

Comment (SpaceIm) [reply recipes/libgeotiff/all/conanfile.py]: @danimtb 

#1529 has been merged, therefore `self.settings.compiler.libcxx` is now removed unconditionally.

**Comment by SpaceIm** [comment]
I've also bumped PROJ dependency version to 7.0.0 (instead of 6.3.1)

File: recipes/libgeotiff/all/patches/fix-cmake.patch [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+-IF(PROJ_FOUND)
+-    INCLUDE_DIRECTORIES(${PROJ_INCLUDE_DIR})
+-ELSE()
+-    MESSAGE(FATAL_ERROR "Failed to detect PROJ >= 6")
+-ENDIF()
+-
+-# Zlib support - optional, default=OFF
+-SET(WITH_ZLIB FALSE CACHE BOOL "Choose if zlib support should be built")
+-
+-IF(WITH_ZLIB)
```
Comment (uilianries) [inline recipes/libgeotiff/all/patches/fix-cmake.patch:44]: Why don't you need zlib and jpeg anymore? Are they required by libtiff only?

Comment (SpaceIm) [reply recipes/libgeotiff/all/patches/fix-cmake.patch:44]: As I said in my first comment, zlib and libjpeg aren't direct dependencies. libgeotiff tries to manage manually dependencies of libtiff.

Comment (uilianries) [reply recipes/libgeotiff/all/patches/fix-cmake.patch:44]: sorry, I missed it (almost 1 month ago). So I think we are good to go.

**Comment by danimtb** [comment]
merging this as it was approved by @jgsogo less than 12 hours ago
